### ISSUE TITLE
Will there be a release the next time (currently 08.2018)?

### ISSUE THREAD

**Issue by rakekniven** [websvnphp/websvn#55]
Hello, just found this fork.
Seen there is some activity.
Will you do a release in the future?

**Comment by michael-o** [comment]
A very good question I wanted to inquire with the other devs. I think nothing holds us back to do so.

**Comment by rakekniven** [comment]
Is current master stable? I have one issue and can test it before release.

**Comment by michael-o** [comment]
Yes, it is. I have been using this for months now. Raise your issue please.

**Comment by michael-o** [comment]
@websvnphp/core-devs I can see that we can create the release directly from GitHub, but how do we change `version.php`?

**Comment by ams-tschoening** [comment]
We could consider not caring about the version info and simply removing it. It doesn't provide that much value to users in my opinion, admins downloading releases as ZIPs get the version info in the file name and those using GIT see the current tag/branch/... at the shell. I prefer everything which makes releases easier, doesn't rely on build tools which need to work cross-platform in theory etc.

Else GitHub should be possible of doing all manual steps online: [Create an new branch](https://help.github.com/articles/creating-and-deleting-branches-within-your-repository/) for the release based on master, e.g. `2.4` this time, change the current view to that branch, navigate to the file, click the file, edit the version number, commit and follow the [release instructions](https://help.github.com/articles/creating-releases/). Afterwards the version number of master needs to be changed as well, can be done online in GitHub again.

![clipboard02](https://user-images.githubusercontent.com/6223655/44340734-0918e700-a486-11e8-9520-da261657bbb8.png)

**Comment by michael-o** [comment]
I desperate miss `mvn release:prepare release:peform`. Sigh...

**Comment by michael-o** [comment]
@ams-tschoening I wouldn't mind to do it manually pretty much like the Maven Release Plugin does. I don't expect us to release weekly.

**Comment by ams-tschoening** [comment]
> Releases are not relying on branches.[...]

One can't commit changes to `version.php` in tags and especially for backporting fixes to releases, creating branches might not be such a bad idea as well.

> I wouldn't mind to do it manually[...]

Feel free to do so, I just collected ideas. :-) I guess we only need the steps I already outlined above and can easily add those to `README.md` to start with. Something like the following:

1. Create branch, e.g. 2.4.
2. Change `version.php` by simply removing `-DEV`-suffix.
3. Push + [release instructions](https://help.github.com/articles/creating-releases/)
4. Increment `version.php` of `master`.

If you are already willing to do the release now, how about adding some small notes to `README.md` as well?

**Comment by miloit** [comment]
When is the release plant will it be also available as dpk for Debian stretch?

**Comment by ams-tschoening** [comment]
> Not that important. Please do a release. Thank you.

Does that mean you error is resolved with a current trunk already or you didn't test for your error at all or ...? Releasing might not make much sense if you already know there's an error which might be easily fixed afterwards. So please test for/provide your problem in some additional issue before a release.

**Comment by rakekniven** [comment]
Hello, not sure this is an error. We will not invest any time at the moment.
For sure I will provide a good report in separate issue if it will be an issue in the future.

**Comment by michael-o** [comment]
I will perform the release now. Let's see how far we get.

**Comment by rakekniven** [comment]
Wouldn't it be better to name it 2.4.0?
Or will the next release be 2.5?

Thanks, will give it a try on monday.

**Comment by rakekniven** [comment]
My vote would be to use semantic versioning.

**Comment by michael-o** [comment]
I agree with that. We will take this into consideration for the next round.

**Comment by miloit** [comment]
Is there any help how to install it via GitHub?

**Comment by michael-o** [comment]
What do you mean to install via GitHub? You need to explode the tarball and config. The documentation is not up to date, granted.

**Comment by ams-tschoening** [comment]
@miloit Your questions get off-topic, please start by reading the available docs, especially because you don't necessarily need to use Linux or GitHub. 

https://websvnphp.github.io/docs/install.html

Afterwards google for the things you don't know about first, because there are dozens of tutorials and step-by-step instructions about basic usage of e.g. Linux and GitHub out there and this issue is not a good place to discuss such things. Of you run into problems which you think have to do with WebSVN itself in your environment, feel free to open new, special issues for those.
